include/lsst/afw/geom/Angle.h

```diff
@@ -246,73 +299,182 @@ Angle const NullAngle = Angle(-1000000., degrees);
 
-template<typename T>
-	double operator /(T const lhs, Angle const rhs);
+/**
+ * Ratio of an angle and a scalar.
+ *
+ * @relatesalso Angle
+ */
+inline constexpr Angle operator/(Angle a, double d) noexcept { return Angle(static_cast<double>(a) / d); }
+
+// Division is different.  Don't allow division by an Angle
+template <typename T>
+constexpr double operator/(T const lhs, Angle rhs) noexcept = delete;
 
-/************************************************************************************************************/
 /**
- * \brief Allow a user to check if they have an angle (yes; they could do this themselves via trivial TMP)
+ * Print an Angle to a stream.
+ *
+ * The exact details of the string representation are unspecified and
+ * subject to change, but the following may be regarded as typical:
+ * `"0.567 rad"`.
+ *
+ * @param s The output stream.
+ * @param a The angle.
+ *
+ * @exceptsafe Provides basic exception guarantee.
+ *
+ * @relatesalso Angle
  */
-template<typename T>
-inline bool isAngle(T) {
-    return false;
-};
+std::ostream& operator<<(std::ostream& s, Angle a);
 
-inline bool isAngle(Angle const&) {
-    return true;
+/************************************************************************************************************/
+
+/// Allow a user to check if they have an angle.
+template <typename T>
+inline constexpr bool isAngle(T) noexcept {
+    return std::is_base_of<Angle, T>::value;
 };
 
 /************************************************************************************************************/
 /**
- * \brief Use AngleUnit to convert a POD (e.g. int, double) to an Angle; e.g. 180*afwGeom::degrees
+ * Use AngleUnit to convert a POD (e.g.\ int, double) to an Angle; e.g.\ 180*::degrees.
+ *
+ * @param lhs the value to convert
+ * @param rhs the conversion coefficient
+ *
+ * @exceptsafe Shall not throw exceptions.
+ *
+ * @relatesalso Angle
  */
-template<typename T>
-inline
-const Angle operator *(T lhs,              ///< the value to convert
-                       AngleUnit const rhs ///< the conversion coefficient
-                      ) {
-    static_assert(std::numeric_limits<T>::is_specialized,
-                            "Only numeric types may be converted to Angles using degrees/radians!");
-    return Angle(lhs*rhs._val);
+template <typename T>
+inline constexpr Angle operator*(T lhs, AngleUnit rhs) noexcept {
+    static_assert(std::is_arithmetic<T>::value,
+                  "Only numeric types may be converted to Angles using degrees/radians!");
+    return Angle(lhs * rhs._val);
+}
+
+/************************************************************************************************************/
+// Inline method definitions, placed last in order to benefit from Angle's full API
+
+inline double Angle::toUnitSphereDistanceSquared() const noexcept {
+    return 2. * (1. - std::cos(asRadians()));
+    // == 4.0 * pow(std::sin(0.5 * asRadians()), 2.0)
```

Comment:
Hmm... both methods are used only in `afw/table/Match.cc`. Perhaps they could at least be moved there, since they aren't really operations on angles in the general sense.